newIDE/app/src/ObjectsList/ObjectRow.js

```diff
@@ -156,11 +158,11 @@ class ThemableObjectRow extends React.Component {
           this.props.onObjectSelected(selected ? '' : objectName);
         }}
         onDoubleClick={(event) => {
-          if (event.button !== 0) return;
-          if (!this.props.onObjectSelected) return;
+          if (event.button !== LEFT_MOUSE_BUTTON) return;
+          if (!this.props.onEdit) return;
           if (this.props.editingName) return;
 
-          this.props.onEdit(selected ? '' : objectName);
+          this.props.onEdit(object);
```

Comment:
onEdit must be called with the object to edit, not the name ;) 
No error was triggered because the file is not flow typed and "onEdit" do not strictly really need the argument to work. Flow typing should ideally be added to improve the robustness of the codebase.